### PR TITLE
Adding Sidecar plugin changes

### PR DESCRIPTION
**Pull request by Nishanthnishu**

### CODE REVIEW

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/SecretsDataHandler.java [hunk 1]
```diff
@@ -48,7 +43,8 @@
 
     @Override
     public List<ManifestSnippet> handle(ServiceSpec serviceSpec, ManifestContext manifestContext) throws HyscaleException {
-        Secrets secrets = SecretsProvider.getSecrets(serviceSpec);
+     //   Secrets secrets = SecretsProvider.getSecrets(serviceSpec);
```
Comment (SameeraGrandhi) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/SecretsDataHandler.java]: Remove these old code Comments

File: service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/model/service/SecretDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.io.IOException;
+
+public class SecretDeserializer extends JsonDeserializer {
+
+    @Override
+    public Object deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
+        JsonNode specNode = jsonParser.readValueAsTree();
+        if(specNode.isArray()){
+            ObjectMapper objectMapper = new ObjectMapper();
```
Comment (tushar-tushar) [inline service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/model/service/SecretDeserializer.java]: To get object mapper you can use ObjectMapperFactory class

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/SecretsDataUtil.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class SecretsDataUtil {
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/SecretsDataUtil.java]: Add java docs

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/AgentVolumesBuilder.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ * </p>
+ *
+ */
+@Component
+public class AgentVolumesBuilder implements AgentBuilder {
+    @Autowired
+    AgentManifestNameGenerator agentManifestNameGenerator;
+
+    @Override
+    public List<ManifestSnippet> build(List<Agent> agents, ServiceSpec serviceSpec) throws JsonProcessingException {
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/AgentVolumesBuilder.java]: Should we wrap the exception and throw HyscaleException?

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/AgentBuilder.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.hyscale.plugin.framework.models.ManifestSnippet;
+import io.hyscale.servicespec.commons.model.service.Agent;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+
+import java.util.List;
+
+public interface AgentBuilder {
+
+    public List<ManifestSnippet> build(List<Agent> agents, ServiceSpec serviceSpec) throws JsonProcessingException;
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/AgentBuilder.java]: Java docs
Should we wrap the exception as HyscaleException?

**Pull request merged**